_guides/managing-oauth-clients.md

```diff
@@ -0,0 +1,209 @@
+---
+title: Managing OAuth Clients
+layout: article
+section: Developing Components
+order: 1
+since: 20190403
+---
+This document reveals the new approach to OAuth utilization within the platform
```

Comment:
don't sure but have idea:
we need to explain what is `Oauth client`, as it's mostly our naming.
So I'd like to say smth like this
1. In some external systems authoirzation is implemented by oauth mechanism.
2. We have components that works with such systems, so we need to be able to use oauth with them
3. Oauth requires some configuration for client (actually this is quite similar to login and password), so platform should provide a way to define some oauth related settings for component
4. We call those settings `oauth client`

Reply:
[OAuth (Open Authorization)](https://en.wikipedia.org/wiki/OAuth) is authorization and authentication protocol based on access tokens. It allows users to forego credential sharing between different secure Internet-based services. Instead it uses automated authorization process that involves dedicated sets of OAuth credentials - "keys" and "secrets".  

Some [integration components](/getting-started/integration-component) use OAuth client to provide accessibility. However, our legacy implementation restricted the user to a single OAuth client per component per [tenant](/getting-started/tenant). The OAuth client for component was defined in the component's environment variables.

For example, for Salesforce component the following environment variables were required in its _component.json_ file:
```
"envVars": {
   "SALESFORCE_KEY": {
     "required": true,
     "description": "Your Salesforce OAuth client key"
   },
   "SALESFORCE_SECRET": {
     "required": true,
     "description": "Your Salesforce OAuth client secret"
   }
},
```
Obviously, `"SALESFORCE_KEY"` defines Oauth key for Salesforce component, and `"SALESFORCE_SECRET"` defines Oauth secret.

This way global access to the component could not be enabled, because users in other tenants would see the
component with its keys and secrets.